lib/components/BucketTaskCard.dart

```diff
@@ -0,0 +1,326 @@
+    final bucket = taskState.buckets[taskState.buckets.indexWhere((b) => b.id == widget.task.bucketId)];
+    // default chip height: 32
+    const double chipHeight = 28;
+    const chipConstraints = BoxConstraints(maxHeight: chipHeight);
+    final theme = Theme.of(context);
+
+    final identifierRow = Row(
+      children: <Widget>[
+        Text(
+          widget.task.identifier.isNotEmpty
+              ? '#${widget.task.identifier.substring(1)}' : '${widget.task.id}',
+          style: theme.textTheme.subtitle2.copyWith(
+            color: Colors.grey,
+          ),
+        ),
+      ],
+    );
+    if (widget.task.done) {
+      identifierRow.children.insert(0, Container(
+        constraints: chipConstraints,
+        padding: EdgeInsets.only(right: 4),
+        child: FittedBox(
+          child: Chip(
+            label: Text('Done'),
+            labelStyle: theme.textTheme.labelLarge.copyWith(
+              fontWeight: FontWeight.bold,
+              color: theme.brightness == Brightness.dark
+                  ? Colors.black : Colors.white,
+            ),
+            backgroundColor: vGreen,
+          ),
+        ),
+      ));
+    }
+
+    final titleRow = Row(
+      children: <Widget>[
+        Expanded(
+          child: Text(
+            widget.task.title,
+            style: theme.textTheme.titleMedium.copyWith(
+              color: widget.task.textColor,
+            ),
+          ),
+        ),
+      ],
+    );
+    if (widget.task.dueDate.year > 2) {
+      final duration = widget.task.dueDate.difference(DateTime.now());
+      final pastDue = duration.isNegative && !widget.task.done;
+      titleRow.children.add(Container(
+        constraints: chipConstraints,
+        padding: EdgeInsets.only(left: 4),
+        child: FittedBox(
+          child: Chip(
+            avatar: Icon(
+              Icons.calendar_month,
+              color: pastDue ? Colors.red : null,
+            ),
+            label: Text(durationToHumanReadable(duration)),
+            labelStyle: theme.textTheme.labelLarge.copyWith(
+              color: pastDue ? Colors.red : null,
+            ),
+            backgroundColor: pastDue ? Colors.red.withAlpha(20) : null,
+          ),
+        ),
+      ));
+    }
+
+    final labelRow = Wrap(
+      children: <Widget>[],
+      spacing: 4,
+      runSpacing: 4,
+    );
+    widget.task.labels?.sort((a, b) => a.title.compareTo(b.title));
+    widget.task.labels?.asMap()?.forEach((i, label) {
+      labelRow.children.add(Chip(
+        label: Text(label.title),
+        labelStyle: theme.textTheme.labelLarge.copyWith(
+          color: label.textColor,
+        ),
+        backgroundColor: label.color,
+      ));
+    });
+    if (widget.task.description.isNotEmpty) {
+      final uncompletedTaskCount = '* [ ]'.allMatches(widget.task.description).length;
+      final completedTaskCount = '* [x]'.allMatches(widget.task.description).length;
```

Comment:
This won't work when the list uses `-` instead of `*`. 
The frontend uses a regex for this: https://kolaente.dev/vikunja/frontend/src/branch/main/src/helpers/checklistFromText.ts

It probably also makes sense to put this in a separate helper function so that we can unit test it on its own.

Reply:
Done, moved to helper functions similar to the frontend implementation in `checkboxes_in_text.dart` which are called from methods in `task.dart` which `BucketTaskCard.dart` now uses.